src/main/java/org/spongepowered/api/map/decoration/MapDecoration.java

```diff
@@ -0,0 +1,163 @@
+     * Gets the position of this MapDecoration, where it is right
+     * now, or where it would be if applied to a {@link org.spongepowered.api.map.MapInfo}.
+     * 0,0 is the centre of a map.
+     * Ranges from {@value java.lang.Byte#MIN_VALUE}-{@value java.lang.Byte#MAX_VALUE}. AKA any valid byte value
+     * @return Vector2i Co-ordinate position in world
+     */
+    Vector2i getPosition();
+
+    /**
+     * Gets the X position on a map, or where it will be when applied
+     * Ranges from {@value java.lang.Byte#MIN_VALUE}-{@value java.lang.Byte#MAX_VALUE}. AKA any valid byte value
+     * @return int Y world coordinate
+     */
+    int getX();
+
+    /**
+     * Gets the Y position on a map. 0 is the center of the map.
+     * Ranges from {@value java.lang.Byte#MIN_VALUE}-{@value java.lang.Byte#MAX_VALUE}. AKA any valid byte value
+     * @return int Y world coordinate
+     */
+    int getY();
+
+    /**
+     * Sets the position of where the MapDecoration is on Maps,
+     * or where it would be if applied to a {@link org.spongepowered.api.map.MapInfo}
+     * 0,0 is the center of the map
+     * Ranges from {@value java.lang.Byte#MIN_VALUE}-{@value java.lang.Byte#MAX_VALUE}. AKA any valid byte value
+     * @param position Vector2i world x and y cords
+     */
+    void setPosition(Vector2i position);
+
+    /**
+     * Sets the X position of where this MapDecoration is on a map
+     * or where it will be when applied
+     * Ranges from {@value java.lang.Byte#MIN_VALUE}-{@value java.lang.Byte#MAX_VALUE}. AKA any valid byte value
+     * @param x world cordinate
+     */
+    void setX(int x);
+
+    /**
+     * Sets the X position of where this MapDecoration is on a map.
+     * 0 is the center of the map.
+     * Ranges from {@value java.lang.Byte#MIN_VALUE}-{@value java.lang.Byte#MAX_VALUE}. AKA any valid byte value
+     * @param y pos
+     */
+    void setY(int y);
+
+    /**
+     * Sets rotation with a Direction. Must be a cardinal/ordinal/secondary ordinal
+     * @param rot Direction, a, secondary ordinal
+     */
+    void setRotation(Direction rot);
+
+    /**
+     * Gets the direction the Map Decoration is pointing
+     * @return Direction, a cardinal/ordinal/secondary
+     */
+    Direction getRotation();
+
+    interface Builder extends DataBuilder<MapDecoration> {
+        /**
+         * Sets the MapDecorationType
+         * @param type MapDecoration to set
+         * @return this Builder, for chaining
+         */
+        Builder type(MapDecorationType type);
+
+        /**
+         * Sets the X position of the MapDecoration
+         * (for when it is applied to a {@link org.spongepowered.api.map.MapInfo})
+         * @param x map coordinate where 0 is the centre of the map.
+         * Ranges from {@value java.lang.Byte#MIN_VALUE}-{@value java.lang.Byte#MAX_VALUE}. AKA any valid byte value
+         * @return this Builder, for chaining
+         */
+        Builder x(int x) throws IllegalStateException;
```

Comment:
This should throw an `IllegalArgumentException` as the error is in what is supplied, not the state of the builder.